pkg/lib/_internal/dnf5daemon.ts

```diff
@@ -297,28 +340,296 @@ export class Dnf5DaemonManager implements PackageManager {
+            case 'transaction_elem_progress': {
+                const [, _last_name, processed,] = args as [string, string, number, number];
+                last_progress = processed / total_packages * 100;
+                break;
+            }
+            }
+
+            if (progress_cb) {
+                progress_cb({
+                    waiting: false,
+                    percentage: last_progress,
+                    cancel: null,
+                });
+            }
         }
 
-        subscription.remove();
-        client.close();
+        await this.with_session(async (session) => {
+            await call(session, "org.rpm.dnf.v0.rpm.Rpm", "remove", [pkgnames, {}]);
+            const [_transaction_items, result] = await call(session, "org.rpm.dnf.v0.Goal", "resolve", [{}]) as RemoveResolveResult;
+            if (result !== 0) {
+                const [problem] = await call(session, "org.rpm.dnf.v0.Goal", "get_transaction_problems_string", []);
+                throw new ResolveError(`Resolving remove failed with result=${result}. ${problem}`);
+            }
+            await call(session, "org.rpm.dnf.v0.Goal", "do_transaction", [{}]);
+        }, signal_emitted);
+    }
+
+    async find_file_packages(files: string[], progress_cb?: ProgressCB): Promise<string[]> {
+        const installed: string[] = [];
+
+        await this.with_session(async (session) => {
+            const package_attrs = ["name"];
+
+            const [results] = await call(session, "org.rpm.dnf.v0.rpm.Rpm", "list", [
+                {
+                    package_attrs: { t: 'as', v: package_attrs },
+                    scope: { t: 's', v: "installed" },
+                    patterns: { t: 'as', v: files },
+                    with_filenames: { t: 'b', v: true },
+                }
+            ]) as ListPackage[][];
+            for (const result of results) {
+                installed.push(result.name.v);
+            }
+
+            // HACK: no usable progress event, but we need to send something to make refresh work.
+            if (progress_cb)
+                progress_cb({ percentage: 100, waiting: false, cancel: null });
+        });
+
+        return installed;
+    }
+
+    async get_updates<T extends boolean>(detail: T, _progress_cb?: ProgressCB): Promise<T extends true ? UpdateDetail[] : Update[]> {
+        const update_map = new Map<string, Update | UpdateDetail>();
+        const package_attrs = ["name", "version", "arch", "epoch", "nevra"];
+
+        await this.with_session(async (session) => {
+            const pkgnames = [];
+            const [results] = await call(session, "org.rpm.dnf.v0.rpm.Rpm", "list", [
+                {
+                    package_attrs: { t: 'as', v: package_attrs },
+                    scope: { t: 's', v: "upgrades" },
+                }
+            ]) as ListPackage[][];
+
+            for (const result of results) {
+                cockpit.assert(result.nevra, "nevra not set");
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22611-966030b0-20251202-151552-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)